### ISSUE TITLE
Speed up build times.

### ISSUE THREAD

**Issue by dannysmc95** [vuejs/vue-cli#6232]
### What problem does this feature solve?
So I have a (not really that large) project in Vue 3 and TypeScript, using the latest versions of the available packages at the time of posting this.

Some details: 
* 46 .vue files.
* 16 .ts files.
* 9 additional vendor packages (fontawesome - fortawesome, lodash.orderby, jquery, c3).

Environment:
* Docker Container
* Image: Ubuntu 20.04.1
* CPUs: 2c/4t
* RAM: 16GB

I am seeing build speeds of around 20 minutes.

This can't be normal? or expected, considering this same project without Vue CLI and in the older vue project takes a minute or 2 max. So what could be happening? There must be something that could be disabled or changed to fix this?

I just can't see Vue staying "ahead" if the build times stay this high, 20 minutes to build for production is pretty bad.
Is there anything we can change?

I have also disabled productionSourceMaps, and skipped the eslint plugin in the build command, but as of now I am out of ideas.
To confirm, I am unable to send you to a reproduction because it's a closed source project for my company - I can answer any questions relating to the project if required though.

### What does the proposed API look like?
N/A

**Comment by dannysmc95** [comment]
Ignore, I spent some time working in the project, looks like it was related to SCSS compilation and fontawesome imports.

**Comment by Anjgpt** [comment]
@dannysmc95 It would be really helpful if you share what exactly you did to reduce the build time.
Thanks in advance.

**Comment by dannysmc95** [comment]
@Anjgpt that's fair.

I mean I have not got the code to hand, but essentially.

So time summary;

Original Compilation - Between 15 - 20 minutes
Updated Compilation - Between 45 seconds - 1 minute and 20 seconds.

(First compilation is usually slower, the first one took around 1 minute and 20 seconds for me then the ones after only took around 45/48 seconds.)

Point 1 - SCSS imports.
I was using a base SCSS folder which included like 6 other SCSS files, 2 of them were just mixins and variables, the other 4 were helpers, and pre-defined styles, and one of the helpers loops and generates like loads of classes, and this was being imported into every single .vue file I had, instead I changed it to only import the two variables and mixins files, and then imported the other 4 files into the main App.vue file instead - this did speed up the SCSS compilation part dramatically.

Point 2: Font Awesome imports.
This was a bit more obvious, but a silly oversight on my end, I was using the @fortawesome package with the SVG fonts that they offered, and using the beta version of the vue-fontawesome component, this was fine, but I was importing the entire font awesome library (over 1600 icons?), instead I imported only the icons I needed which was about 30/40 icons. This sped up the compilation process massively, and also lowered the vue webpack dev server initial's download, which was originally around 58MB to now 16MB.

TL;DR - Import only the fonts you need and make sure, when using fortawesome and SVG icons, do not import the entire icon library, and instead import only the icons you need. If you use like base SCSS sheets and you use scoped styling for your components, only import variables and mixins to each Vue file, and load helpers and any looping SCSS code globally to help with speeds.

**Comment by Anjgpt** [comment]
@dannysmc95 Thanks for the detailed summary.
